src/coreclr/vm/interpreter.cpp

```diff
@@ -10851,6 +10870,46 @@ void Interpreter::DoGetIsSupported()
     m_curStackHt++;
 }
 
+void Interpreter::DoGetArrayDataReference()
+{
+    CONTRACTL {
+            THROWS;
+            GC_TRIGGERS;
+            MODE_COOPERATIVE;
+        } CONTRACTL_END;
+
+    _ASSERTE(m_curStackHt > 0);
+    unsigned ind = m_curStackHt - 1;
+
+#ifdef _DEBUG
+    CorInfoType stringCIT = OpStackTypeGet(ind).ToCorInfoType();
+    if (stringCIT != CORINFO_TYPE_CLASS)
+    {
+        VerificationError("DoGetArrayDataReference called on non-array.");
+    }
```

Comment:
```suggestion
    _ASSERTE(OpStackTypeGet(ind).ToCorInfoType() == CORINFO_TYPE_CLASS);
```
This should be assert. We do not want to have behavior differences between debug and release builds.